gson/src/test/java/com/google/gson/ConstructorContractTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2011 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gson;
+
+import com.google.gson.common.MoreAsserts;
+import com.google.gson.internal.$Gson$Preconditions;
+import com.google.gson.internal.$Gson$Types;
+import com.google.gson.internal.Primitives;
+import com.google.gson.internal.Streams;
+import com.google.gson.internal.bind.TypeAdapters;
+import junit.framework.TestCase;
+
+/**
+ * Unit test to check constructors in util classes.
+ *
+ * @author Vladislav Bauer
+ */
+
+public class ConstructorContractTest extends TestCase {
```

Comment:
Yes. But this test offers extremely little utility: it's unlikely to ever detect a regression. And the severity of that bug is also extremely low. We want to write tests that prevent problems. This doesn't do that.